custom_components/tahoma/sensor.py

```diff
@@ -100,12 +100,17 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
+        attribute = [
+            attr.value
+            for attr in self.device.attributes
+            if attr.name == CORE_MEASURED_VALUE_TYPE
+        ]
         TEMPERATURE_UNIT = {
             CORE_TEMPERATURE_IN_CELSIUS: TEMP_CELSIUS,
             CORE_TEMPERATURE_IN_CELCIUS: TEMP_CELSIUS,
             CORE_TEMPERATURE_IN_KELVIN: TEMP_KELVIN,
             CORE_TEMPERATURE_IN_FAHRENHEIT: TEMP_FAHRENHEIT,
-        }.get(self.device.attributes.get(CORE_MEASURED_VALUE_TYPE), TEMP_CELSIUS,)
+        }.get(attribute[0], TEMP_CELSIUS,)
```

Comment:
@vlebourl should we just use attribute.value here? That makes it more readable and nicely typed. Just use `attr` in the loop.

Reply:
except `attribute` is a list, it doesn't have a `.value` attribute.

Reply:
` attribute = [
            attr.value
            for attr in self.device.attributes
            if attr.name == CORE_MEASURED_VALUE_TYPE
        ]
```

Attribute is not a list anymore? Otherwise we should change something in the loop, to just get the state back, so you can do attribute.value. Because taking the first element of a list doesn't sound safe to me?

Reply:
I think I will introduce a class States in the python-tahoma-api to add a getter method. We repeat a lot this pattern.

Reply:
@tetienne agree, this is something we should solve in the wrapper. Shall we merge this in for now?